### PR TITLE
Test (qcdq-qop): Expanded tests with pytest_cases

### PR DESCRIPTION
**Pull request by Giuseppe5**
Follow up on #431

### CODE REVIEW

File: tests/brevitas_ort/common.py [hunk 1]
```diff
@@ -3,6 +3,7 @@
 import torch
 import onnxruntime as ort
 import numpy as np
+from brevitas import export
```
Comment (Giuseppe5) [inline tests/brevitas_ort/common.py:6]: To be removed

File: tests/brevitas_ort/common.py [hunk 2]
```diff
@@ -17,11 +18,10 @@
 OUT_CH = 16
 IN_CH = 8
 FEATURES = 5
-TOLERANCE = 1  # accept +1/-1 errors
+TOLERANCE = True
 KERNEL_SIZE = 1  # keep float error during fake-quantization under control
 BIT_WIDTHS = range(2, 9)
 QUANTIZERS = {
- #   'asymmetric_float': (ShiftedUint8WeightPerTensorFloat, ShiftedUint8ActPerTensorFloat),
```
Comment (volcacius) [inline tests/brevitas_ort/common.py:24]: Enable back, not remove

File: tests/brevitas_ort/test_quant_module.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+from pytest_cases import parametrize_with_cases, get_case_id
+
+@parametrize_with_cases('model', cases=QuantWBIOLCases)
+@pytest.mark.parametrize('export_type', ['qop', 'qcdq'])
+def test_ort(model, export_type, current_cases):
+    cases_generator_func = current_cases['model'][1]
+    case_id = get_case_id(cases_generator_func)
+    impl = case_id.split('-')[-2] # Inverse list of definition, 'export_type' is -1, 'impl' is -2, etc.
+
+    if impl in ('QuantConvTranspose1d', 'QuantConvTranspose2d'):
```
Comment (volcacius) [inline tests/brevitas_ort/test_quant_module.py:21]: QuantConvTranspose is supposed to work on qcdq

File: tests/brevitas_ort/common.py [hunk 4]
```diff
@@ -35,7 +35,7 @@ def compute_ort(export_name, np_input):
     return pred_onx
 
 
-def is_brevitas_ort_close(model, np_input, export_name, export_type, tolerance=None):
+def is_brevitas_ort_close(model, np_input, export_name, export_type, tolerance_flag=False):
```
Comment (volcacius) [inline tests/brevitas_ort/common.py:38]: Keep tolerance=None, pass in a constant as before

File: tests/brevitas_ort/test_quant_module.py [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+        pytest.skip('Export of ConvTranspose is not supported')
+
+    if impl in ('QuantLinear'):
+        IN_SIZE = (1, IN_CH)
```
Comment (volcacius) [inline tests/brevitas_ort/test_quant_module.py:25]: in_size =

File: tests/brevitas_ort/test_quant_module.py (hunk 3 above)
Comment (Giuseppe5) [reply tests/brevitas_ort/test_quant_module.py:21]: What about QOp?

Comment (volcacius) [reply tests/brevitas_ort/test_quant_module.py:21]: there is no quantized transposed conv in in the standard onnx opset

File: tests/brevitas_ort/test_quant_module.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+from .common import *
+from pytest_cases import parametrize_with_cases
+from .quant_module_cases import QuantWBIOLCases
+from pytest_cases import parametrize_with_cases, get_case_id
+from tests.marker import requires_pt_ge
+
+@parametrize_with_cases('model', cases=QuantWBIOLCases)
+@pytest.mark.parametrize('export_type', ['qcdq', 'qop'])
+@requires_pt_ge('1.8.1')
```
Comment (Giuseppe5) [inline tests/brevitas_ort/test_quant_module.py:15]: @volcacius To support bitwidth<8 and narrow_range quantization (enabled by default in weight quantization), we need Clip operation, which has been introduced only from pytorch 1.8.1. Should we raise an error like we do for the axis flag in per_channel quantization?

**Review by volcacius — CHANGES_REQUESTED** [review]
This should be rebased into two commits, one with the workaround for the zero-point, the rest with the updated tests. Anything related to CI (removing support for 3.6, pull request on all branches) should be removed.

**Comment by Giuseppe5** [comment]
All done, let me know if there is more clean-up to do

**Review by volcacius — APPROVED** [review]
